packages/arb-token-bridge-ui/src/state/app/utils.ts

```diff
@@ -307,15 +305,34 @@ export const isDepositReadyToRedeem = (tx: MergedTransaction) => {
   return isDeposit(tx) && tx.depositStatus === DepositStatus.L2_FAILURE
 }
 
-export const getStandardizedTimestamp = (date: string | BigNumber) => {
+export const normalizeTimestamp = (date: number | string | BigNumber) => {
   // because we get timestamps in different formats from subgraph/event-logs/useTxn hook, we need 1 standard format.
+  const TIMESTAMP_LENGTH = 13
+  let timestamp
+
   if (typeof date === 'string') {
-    if (isNaN(Number(date))) return dayjs(new Date(date)).unix() // for ISOstring type of dates -> dayjs timestamp
-    return Number(date) // for timestamp type of date -> dayjs timestamp
+    timestamp = isNaN(Number(date))
+      ? dayjs(new Date(date)).unix() // for ISOstring type of dates -> dayjs timestamp
+      : Number(date) // for timestamp type of date -> dayjs timestamp
+  } else if (typeof date !== 'number') {
+    // BigNumber
+    timestamp = date.toNumber()
+  }
+
+  const timestampString = String(timestamp)
+
+  if (timestampString.length === TIMESTAMP_LENGTH) {
+    // correct timestamp length
+    return Math.floor(Number(timestampString))
```

Comment:
Why is `Math.floor` needed for timestamps? 🤔

Reply:
https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1868/files#diff-286a4ee4334ef698f2f4080f95722061ff721facd19401fdd32279bf686c1e91L935

We did it here before, I guess it's just in case because we divide by 1000 we want to make sure it's always in integer?

Reply:
but we don't do division by 1000 now, why is `Math.floor` still needed?

Reply:
updated